neural_lam/vis.py

```diff
@@ -179,3 +181,90 @@ def plot_spatial_error(error, title=None, vrange=None):
         fig.suptitle(title, size=10)
 
     return fig
+
+@matplotlib.rc_context(utils.fractional_plot_bundle(1))
+def verify_inference(file_path:str, feature_channel:int, vrange=None, save_path=None):
+    """
+    Plot example prediction, forecast, and ground truth.
+    Each has shape (N_grid,)
+    """
+
+    # Verify that feature channel is between 0 and 42
+    if not 0 <= feature_channel <= 42:
+        raise ValueError("feature_channel must be between 0 and 42, inclusive.")
+
+    # Load the inference dataset for plotting
+    predictions_data_module = WeatherDataModule(
+            "cosmo",
+            path_verif_file=file_path,
+            split="verification",
+            standardize=False,
+            subset=False,
+            batch_size=6,
+            num_workers=2
+        )
+    predictions_data_module.setup(stage='verification')
+    predictions_loader = predictions_data_module.verification_dataloader() 
+    for predictions_batch in predictions_loader:
+        predictions = predictions_batch[0]  # tensor 
+        break 
+
+    # get test data
+    data_latlon = xr.open_zarr(constants.EXAMPLE_FILE).isel(time=0)
+    lon, lat = unrotate_latlon(data_latlon)
+
+    # Get common scale for values
+    total =  predictions[0,:,:,feature_channel]
+    total_array = np.array(total)
+    if vrange is None:
+        vmin = total_array.min()
+        vmax = total_array.max()
+    else:
+        vmin, vmax = float(vrange[0].cpu().item()), float(vrange[1].cpu().item())
+
+    # Plot
+    for i in range(23):
```

Comment:
Also here 23 is a value defined by the user in constants.py called `eval_horizon -2`. This loop should be more flexible and take the user input into account